# ANALYSIS

# Aim:      
        To find the ranges of Hyperparamteres for every model of each runtime

In [ ]:
import numpy as np
import pandas as pd

### DRF ANALYSIS

In [10]:
df_drf=pd.read_csv("DRF_HP.csv", decimal = ',')

In [11]:
df_drf.head(10)

Model_name fold_assignment  fold_column  offset_column  \
0  DRF_1_AutoML_20190420_203958          Modulo          NaN            NaN   
1  DRF_1_AutoML_20190420_204019          Modulo          NaN            NaN   
2  DRF_1_AutoML_20190420_204025          Modulo          NaN            NaN   
3  DRF_1_AutoML_20190419_152459          Modulo          NaN            NaN   
4  DRF_1_AutoML_20190421_181829          Modulo          NaN            NaN   
5  DRF_1_AutoML_20190421_183436          Modulo          NaN            NaN   
6  DRF_1_AutoML_20190421_184305          Modulo          NaN            NaN   
7  DRF_1_AutoML_20190421_184315          Modulo          NaN            NaN   

   weights_column  balance_classes  class_sampling_factors  \
0             NaN            False                     NaN   
1             NaN            False                     NaN   
2             NaN            False                     NaN   
3             NaN            False                     NaN   
4             NaN            False                     NaN   
5             NaN            False                     NaN   
6             NaN            False                     NaN   
7             NaN            False                     NaN   

   max_after_balance_size  ntrees  max_depth  ...  max_runtime_secs  \
0                       5      15         20  ...                 0   
1                       5      15         20  ...                 0   
2                       5      16         20  ...                 0   
3                       5      15         20  ...                 0   
4                       5      12         20  ...                 0   
5                       5      27         20  ...                 0   
6                       5      15         20  ...                 0   
7                       5      14         20  ...                 0   

        seed  mtries  sample_rate  sample_rate_per_class  \
0  -7.32E+18      -1  0.632000029                    NaN   
1   6.41E+18      -1  0.632000029                    NaN   
2   3.62E+18      -1  0.632000029                    NaN   
3  -8.38E+18      -1  0.632000029                    NaN   
4  -3.28E+18      -1  0.632000029                    NaN   
5  -7.76E+18      -1  0.632000029                    NaN   
6   9.08E+18      -1  0.632000029                    NaN   
7  -2.01E+18      -1  0.632000029                    NaN   

  col_sample_rate_change_per_level col_sample_rate_per_tree  \
0                                1                        1   
1                                1                        1   
2                                1                        1   
3                                1                        1   
4                                1                        1   
5                                1                        1   
6                                1                        1   
7                                1                        1   

   min_split_improvement histogram_type  categorical_encoding  
0               1.00E-05           AUTO                  AUTO  
1               1.00E-05           AUTO                  AUTO  
2               1.00E-05           AUTO                  AUTO  
3               1.00E-05           AUTO                  AUTO  
4               1.00E-05           AUTO                  AUTO  
5               1.00E-05           AUTO                  AUTO  
6               1.00E-05           AUTO                  AUTO  
7               1.00E-05           AUTO                  AUTO  

[8 rows x 27 columns]

In [5]:
df_drf.dtypes

Model_name                           object
fold_assignment                      object
fold_column                         float64
offset_column                       float64
weights_column                      float64
balance_classes                        bool
class_sampling_factors              float64
max_after_balance_size                int64
ntrees                                int64
max_depth                             int64
min_rows                              int64
nbins                                 int64
nbins_top_level                       int64
nbins_cats                            int64
stopping_rounds                       int64
stopping_metric                      object
stopping_tolerance                   object
max_runtime_secs                      int64
seed                                 object
mtries                                int64
sample_rate                          object
sample_rate_per_class               float64
col_sample_rate_change_per_level

# Hyperparameter Ranges

In [12]:
range_mtries = [df_drf['mtries'].min(),df_drf['mtries'].max()]
range_mtries

[-1, -1]

In [13]:
ntrees = [df_drf['ntrees'].min(),df_drf['ntrees'].max()]
ntrees

[12, 27]

In [14]:
seed = [df_drf['seed'].min(),df_drf['seed'].max()]
seed

['-2.01E+18', '9.08E+18']

|Hyperparameter| Minimum | Maximum |
|---|---|---|
|seed|-2.01E+18 | 9.08E+18 |
|ntrees|12| 27|
|range_metries|-1| -1|

### GBM ANALYSIS

In [17]:
df_gbm=pd.read_csv("GBM_HP.csv", decimal = ',')

In [48]:
df_gbm.head()

Model_name  Unnamed: 2  offset_column  weights_column  \
0  GBM_1_AutoML_20190420_203958         NaN            NaN             NaN   
1  GBM_1_AutoML_20190420_204019         NaN            NaN             NaN   
2  GBM_1_AutoML_20190420_204025         NaN            NaN             NaN   
3  GBM_2_AutoML_20190420_203958         NaN            NaN             NaN   
4  GBM_2_AutoML_20190420_204025         NaN            NaN             NaN   

   class_sampling_factors  max_after_balance_size  ntrees  min_rows  nbins  \
0                     NaN                       5      30       100     20   
1                     NaN                       5      32       100     20   
2                     NaN                       5      30       100     20   
3                     NaN                       5      30       100     20   
4                     NaN                       5      30       100     20   

   nbins_top_level  ...  learn_rate  learn_rate_annealing sample_rate  \
0             1024  ...         0.1                     1         0.8   
1             1024  ...         0.1                     1         0.8   
2             1024  ...         0.1                     1         0.8   
3             1024  ...         0.1                     1         0.8   
4             1024  ...         0.1                     1         0.8   

  sample_rate_per_class col_sample_rate col_sample_rate_change_per_level  \
0                   NaN             0.8                                1   
1                   NaN             0.8                                1   
2                   NaN             0.8                                1   
3                   NaN             0.8                                1   
4                   NaN             0.8                                1   

   col_sample_rate_per_tree min_split_improvement    max_abs_leafnode_pred  \
0                       0.8              1.00E-05  1.7976931348623151e+308   
1                       0.8              1.00E-05  1.7976931348623151e+308   
2                       0.8              1.00E-05  1.7976931348623151e+308   
3                       0.8              1.00E-05  1.7976931348623151e+308   
4                       0.8              1.00E-05  1.7976931348623151e+308   

  pred_noise_bandwidth  
0                    0  
1                    0  
2                    0  
3                    0  
4                    0  

[5 rows x 25 columns]

In [49]:
df_gbm.dtypes

Model_name                           object
Unnamed: 2                          float64
offset_column                       float64
weights_column                      float64
class_sampling_factors              float64
max_after_balance_size                int64
ntrees                                int64
min_rows                              int64
nbins                                 int64
nbins_top_level                       int64
nbins_cats                            int64
stopping_rounds                       int64
stopping_tolerance                   object
max_runtime_secs                     object
seed                                 object
learn_rate                           object
learn_rate_annealing                  int64
sample_rate                          object
sample_rate_per_class               float64
col_sample_rate                      object
col_sample_rate_change_per_level      int64
col_sample_rate_per_tree             object
min_split_improvement           

Dropping the colums with the values which are not responsible to give any range 

In [50]:
df_gbm['distribution'].head()

KeyError: 'distribution'

In [36]:
df_gbm['fold_assignment'].head()

0    Modulo
1    Modulo
2    Modulo
3    Modulo
4    Modulo
Name: fold_assignment, dtype: object

In [37]:
df_gbm['stopping_metric'].head()

0    logloss
1    logloss
2    logloss
3    logloss
4    logloss
Name: stopping_metric, dtype: object

In [38]:
df_gbm['histogram_type'].head()

0    AUTO
1    AUTO
2    AUTO
3    AUTO
4    AUTO
Name: histogram_type, dtype: object

In [39]:
df_gbm= df_gbm.drop(columns=['distribution','fold_assignment','histogram_type','stopping_metric','Model_name'])
df_gbm

Model_name  Unnamed: 2  offset_column  \
0                  GBM_1_AutoML_20190420_203958         NaN            NaN   
1                  GBM_1_AutoML_20190420_204019         NaN            NaN   
2                  GBM_1_AutoML_20190420_204025         NaN            NaN   
3                  GBM_2_AutoML_20190420_203958         NaN            NaN   
4                  GBM_2_AutoML_20190420_204025         NaN            NaN   
5                  GBM_3_AutoML_20190420_203958         NaN            NaN   
6                  GBM_3_AutoML_20190420_204025         NaN            NaN   
7                  GBM_4_AutoML_20190420_203958         NaN            NaN   
8                  GBM_4_AutoML_20190420_204025         NaN            NaN   
9                  GBM_5_AutoML_20190420_203958         NaN            NaN   
10                 GBM_5_AutoML_20190420_204025         NaN            NaN   
11   GBM_grid_1_AutoML_20190420_203958_model_10         NaN            NaN   
12   GBM_grid_1_AutoML_20190420_203958_model_11         NaN            NaN   
13   GBM_grid_1_AutoML_20190420_203958_model_12         NaN            NaN   
14   GBM_grid_1_AutoML_20190420_203958_model_13         NaN            NaN   
15   GBM_grid_1_AutoML_20190420_203958_model_14         NaN            NaN   
16   GBM_grid_1_AutoML_20190420_203958_model_15         NaN            NaN   
17    GBM_grid_1_AutoML_20190420_203958_model_1         NaN            NaN   
18    GBM_grid_1_AutoML_20190420_203958_model_2         NaN            NaN   
19    GBM_grid_1_AutoML_20190420_203958_model_3         NaN            NaN   
20    GBM_grid_1_AutoML_20190420_203958_model_4         NaN            NaN   
21    GBM_grid_1_AutoML_20190420_203958_model_5         NaN            NaN   
22    GBM_grid_1_AutoML_20190420_203958_model_6         NaN            NaN   
23    GBM_grid_1_AutoML_20190420_203958_model_7         NaN            NaN   
24    GBM_grid_1_AutoML_20190420_203958_model_8         NaN            NaN   
25    GBM_grid_1_AutoML_20190420_203958_model_9         NaN            NaN   
26   GBM_grid_1_AutoML_20190420_204025_model_10         NaN            NaN   
27    GBM_grid_1_AutoML_20190420_204025_model_1         NaN            NaN   
28    GBM_grid_1_AutoML_20190420_204025_model_2         NaN            NaN   
29    GBM_grid_1_AutoML_20190420_204025_model_3         NaN            NaN   
..                                          ...         ...            ...   
104   GBM_grid_1_AutoML_20190421_183436_model_7         NaN            NaN   
105   GBM_grid_1_AutoML_20190421_183436_model_8         NaN            NaN   
106   GBM_grid_1_AutoML_20190421_183436_model_9         NaN            NaN   
107                GBM_1_AutoML_20190421_184305         NaN            NaN   
108                GBM_1_AutoML_20190421_184315         NaN            NaN   
109                GBM_2_AutoML_20190421_184305         NaN            NaN   
110                GBM_2_AutoML_20190421_184315         NaN            NaN   
111                GBM_3_AutoML_20190421_184305         NaN            NaN   
112                GBM_3_AutoML_20190421_184315         NaN            NaN   
113                GBM_4_AutoML_20190421_184315         NaN            NaN   
114                GBM_5_AutoML_20190421_184315         NaN            NaN   
115  GBM_grid_1_AutoML_20190421_184315_model_10         NaN            NaN   
116  GBM_grid_1_AutoML_20190421_184315_model_11         NaN            NaN   
117  GBM_grid_1_AutoML_20190421_184315_model_12         NaN            NaN   
118  GBM_grid_1_AutoML_20190421_184315_model_13         NaN            NaN   
119  GBM_grid_1_AutoML_20190421_184315_model_14         NaN            NaN   
120  GBM_grid_1_AutoML_20190421_184315_model_15         NaN            NaN   
121  GBM_grid_1_AutoML_20190421_184315_model_16         NaN            NaN   
122  GBM_grid_1_AutoML_20190421_184315_model_17         NaN            NaN   
123  GBM_grid_1_AutoML_20190421_184315_model_18         NaN

In [51]:
df_gbm= df_gbm.drop(columns=['Model_name'])
df_gbm

Unnamed: 2  offset_column  weights_column  class_sampling_factors  \
0           NaN            NaN             NaN                     NaN   
1           NaN            NaN             NaN                     NaN   
2           NaN            NaN             NaN                     NaN   
3           NaN            NaN             NaN                     NaN   
4           NaN            NaN             NaN                     NaN   
5           NaN            NaN             NaN                     NaN   
6           NaN            NaN             NaN                     NaN   
7           NaN            NaN             NaN                     NaN   
8           NaN            NaN             NaN                     NaN   
9           NaN            NaN             NaN                     NaN   
10          NaN            NaN             NaN                     NaN   
11          NaN            NaN             NaN                     NaN   
12          NaN            NaN             NaN                     NaN   
13          NaN            NaN             NaN                     NaN   
14          NaN            NaN             NaN                     NaN   
15          NaN            NaN             NaN                     NaN   
16          NaN            NaN             NaN                     NaN   
17          NaN            NaN             NaN                     NaN   
18          NaN            NaN             NaN                     NaN   
19          NaN            NaN             NaN                     NaN   
20          NaN            NaN             NaN                     NaN   
21          NaN            NaN             NaN                     NaN   
22          NaN            NaN             NaN                     NaN   
23          NaN            NaN             NaN                     NaN   
24          NaN            NaN             NaN                     NaN   
25          NaN            NaN             NaN                     NaN   
26          NaN            NaN             NaN                     NaN   
27          NaN            NaN             NaN                     NaN   
28          NaN            NaN             NaN                     NaN   
29          NaN            NaN             NaN                     NaN   
..          ...            ...             ...                     ...   
104         NaN            NaN             NaN                     NaN   
105         NaN            NaN             NaN                     NaN   
106         NaN            NaN             NaN                     NaN   
107         NaN            NaN             NaN                     NaN   
108         NaN            NaN             NaN                     NaN   
109         NaN            NaN             NaN                     NaN   
110         NaN            NaN             NaN                     NaN   
111         NaN            NaN             NaN                     NaN   
112         NaN            NaN             NaN                     NaN   
113         NaN            NaN             NaN                     NaN   
114         NaN            NaN             NaN                     NaN   
115         NaN            NaN             NaN                     NaN   
116         NaN            NaN             NaN                     NaN   
117         NaN            NaN             NaN                     NaN   
118         NaN            NaN             NaN                     NaN   
119         NaN            NaN             NaN                     NaN   
120         NaN            NaN             NaN                     NaN   
121         NaN            NaN             NaN                     NaN   
122         NaN            NaN             NaN                     NaN   
123         NaN            NaN             NaN                     NaN   
124         NaN            NaN             NaN                     NaN   
125         NaN            NaN             NaN                     NaN   
126      

In [52]:
ranges = []
for key,val in df_gbm.dtypes.items():
    print(key,val)
    if val == 'int64': 
         ranges.append([key,[df_gbm[key].min(),df_gbm[key].max()]])
    ##if val == 'float64':
      ##   ranges.append([key,[df_gbm[key].min(),df_gbm[key].max()]])
    if val == 'object':
         ranges.append([key,[df_gbm[key].min(),df_gbm[key].max()]])

Unnamed: 2 float64
offset_column float64
weights_column float64
class_sampling_factors float64
max_after_balance_size int64
ntrees int64
min_rows int64
nbins int64
nbins_top_level int64
nbins_cats int64
stopping_rounds int64
stopping_tolerance object
max_runtime_secs object
seed object
learn_rate object
learn_rate_annealing int64
sample_rate object
sample_rate_per_class float64
col_sample_rate object
col_sample_rate_change_per_level int64
col_sample_rate_per_tree object
min_split_improvement object
max_abs_leafnode_pred object
pred_noise_bandwidth int64


# Hyperparameter Ranges

In [53]:
ranges

[['max_after_balance_size', [5, 5]],
 ['ntrees', [30, 43]],
 ['min_rows', [1, 100]],
 ['nbins', [20, 20]],
 ['nbins_top_level', [1024, 1024]],
 ['nbins_cats', [1024, 1024]],
 ['stopping_rounds', [0, 3]],
 ['stopping_tolerance', ['0.041415768', '0.041415768']],
 ['max_runtime_secs', ['0', '293.974']],
 ['seed', ['-1.37645E+18', '9.15531E+18']],
 ['learn_rate', ['0.001', '0.8']],
 ['learn_rate_annealing', [1, 1]],
 ['sample_rate', ['0.5', '1']],
 ['col_sample_rate', ['0.4', '1']],
 ['col_sample_rate_change_per_level', [1, 1]],
 ['col_sample_rate_per_tree', ['0.4', '1']],
 ['min_split_improvement', ['0.0001', '1.00E-05']],
 ['max_abs_leafnode_pred',
  ['1.7976931348623151e+308', '1.7976931348623151e+308']],
 ['pred_noise_bandwidth', [0, 0]]]

|Hyperparameter| Minimum | Maximum |
|---|---|---|
|ntrees|30|43|
|min_rows|1|100|
|learn_rate|0.001|0.8|
|learn_rate_annealing|1|1|
|col_sample_rate|0.4|1|
|sample_rate|0.5| 1.0|
|col_sample_rate_per_tree|0.4| 1.0|
|max_abs_leafnode_pred|1.7976931348623157e+308|1.7976931348623157e+308|
|pred_noise_bandwidth|0|0|

### GLM ANALYSIS

In [56]:
df_glm=pd.read_csv("GLM_HP.csv", decimal = ',')

In [57]:
df_glm.head()

Model_Name  Weights_Column  Offset_Column  \
0  GLM_grid_1_AutoML_20190420_203958_model_1             NaN            NaN   
1  GLM_grid_1_AutoML_20190420_204019_model_1             NaN            NaN   
2  GLM_grid_1_AutoML_20190420_204025_model_1             NaN            NaN   
3  GLM_grid_1_AutoML_20190421_181829_model_1             NaN            NaN   
4  GLM_grid_1_AutoML_20190421_183436_model_1             NaN            NaN   

   FoldColumn Fold_Assignment          Seed  tweedie_variance_power  \
0         NaN          Modulo  -4.60175E+18                       0   
1         NaN          Modulo   3.10247E+17                       0   
2         NaN          Modulo   2.38493E+18                       0   
3         NaN          Modulo  -3.64012E+17                       0   
4         NaN          Modulo   5.38318E+18                       0   

   tweedie_link_power\t                                          Alpha  \
0                     1  [0.0, 0.2, 0.4, 0.6000000000000001, 0.8, 1.0]   
1                     1  [0.0, 0.2, 0.4, 0.6000000000000001, 0.8, 1.0]   
2                     1  [0.0, 0.2, 0.4, 0.6000000000000001, 0.8, 1.0]   
3                     1  [0.0, 0.2, 0.4, 0.6000000000000001, 0.8, 1.0]   
4                     1  [0.0, 0.2, 0.4, 0.6000000000000001, 0.8, 1.0]   

                                              Lambda Missing_Values_Handling  \
0  [11.114257974474292, 6.9021424987019255, 4.286...          MeanImputation   
1  [11.114257974474292, 6.9021424987019255, 4.286...          MeanImputation   
2  [11.114257974474292, 6.9021424987019255, 4.286...          MeanImputation   
3  [11.114257974474292, 6.9021424987019255, 4.286...          MeanImputation   
4  [11.114257974474292, 6.9021424987019255, 4.286...          MeanImputation   

   Standardize     theta  
0         True  1.00E-10  
1         True  1.00E-10  
2         True  1.00E-10  
3         True  1.00E-10  
4         True  1.00E-10

# Hyperparameters Ranges

In [59]:
alpha_list = ['0.0', '0.2', '0.4', '0.600000000000000001', '0.8', '1.0']
alpha = [float(i) for i in alpha_list] 
alpha

[0.0, 0.2, 0.4, 0.6, 0.8, 1.0]

In [60]:
halpha = [min(alpha),max(alpha)]
halpha

[0.0, 1.0]

In [62]:
lambda_list = ['11.114257974474292', '6.9021424987019255', '4.286347427043651', '2.6618943710274108', '1.653081501933804', '1.026591619028441', '0.6375307878205491', '0.39591742021403503', '0.24587142554291602', '0.15269032078918002', '0.094823276072938', '0.058886860926955005', '0.036569738290451', '0.022710427038911', '0.014103559948755002', '0.008758549669160001', '0.0054392077309470006', '0.0033778401513850005', '0.002097695961012', '0.0013027047307260001', '0.0008090017080630001', '0.0005024037667260001', '0.000312001250807']
lambda_list = [float(i) for i in lambda_list] 
lambda_list

[11.114257974474292,
 6.9021424987019255,
 4.286347427043651,
 2.6618943710274108,
 1.653081501933804,
 1.026591619028441,
 0.6375307878205491,
 0.39591742021403503,
 0.24587142554291602,
 0.15269032078918002,
 0.094823276072938,
 0.058886860926955005,
 0.036569738290451,
 0.022710427038911,
 0.014103559948755002,
 0.008758549669160001,
 0.0054392077309470006,
 0.0033778401513850005,
 0.002097695961012,
 0.0013027047307260001,
 0.0008090017080630001,
 0.0005024037667260001,
 0.000312001250807]

In [63]:
hlambda = [min(lambda_list),max(lambda_list)]
hlambda

[0.000312001250807, 11.114257974474292]

In [65]:
hseed = [df_glm['Seed'].min(),df_glm['Seed'].max()]
hseed

['-3.18782E+18', '5.38318E+18']

In [66]:
htv = [df_glm['tweedie_variance_power'].min(),df_glm['tweedie_variance_power'].max()]
htv

[0, 0]

In [68]:
theta = [df_glm['theta'].min(),df_glm['theta'].max()]
theta

['1.00E-10', '1.00E-10']

|Hyperparameter| Minimum | Maximum |
|---|---|---|
|seed|-3.18782E+18 | 5.38318E+18|
|tweedie_variance_power|0.0 | 0.0|
|alpha|0.0 | 1.0|
|lambda_list|0.000312001250807| 11.114257974474292|
|theta|1.00E-10| 1.00E-10|

# XRT ANALYSIS

In [70]:
df_xrt=pd.read_csv("XRT_HP.csv", decimal = ',')

In [71]:
df_xrt.head()

Model_Name  Weights_Column  Offset_Column  \
0   XRT_1_AutoML_20190421_181829__800             NaN            NaN   
1   XRT_1_AutoML_20190420_203958__300             NaN            NaN   
2   XRT_1_AutoML_20190420_204025__300             NaN            NaN   
3  XRT_1_AutoML_20190421_183436__1000             NaN            NaN   
4   XRT_1_AutoML_20190419_152459__500             NaN            NaN   

   Fold_Column Fold_Assignment  Stopping_Rounds  max_runtime_secs  \
0          NaN          Modulo                0                 0   
1          NaN          Modulo                0                 0   
2          NaN          Modulo                0                 0   
3          NaN          Modulo                0                 0   
4          NaN          Modulo                0                 0   

  Stopping_metric Stopping_tolerance  Balance_Classes  ...  nbins_cats  \
0         Logloss        0.041415768            False  ...        1024   
1         Logloss        0.041415768            False  ...        1024   
2         Logloss        0.041415768            False  ...        1024   
3         Logloss        1.041415768            False  ...        1024   
4         Logloss        0.041415768            False  ...        1024   

           seed  Sample_Rate  Sample_rate_per_class  Col_sample_rate_per_tree  \
0   4.84711E+18  0.632000029                    NaN                         1   
1   8.79434E+18  0.632000029                    NaN                         1   
2  -8.85259E+18  0.632000029                    NaN                         1   
3  -8.85259E+18  0.632000029                    NaN                         1   
4  -8.85259E+18  0.632000029                    NaN                         1   

   Col_sample_rate_Change_per_level  min_split_improvement  histogram_type  \
0                                 1               1.00E-05          RANDOM   
1                                 1               1.00E-05          RANDOM   
2                                 1               1.00E+00          RANDOM   
3                                 1               1.00E+00          RANDOM   
4                                 1               1.00E+00          RANDOM   

  mtries categorical_encoding  
0     -1                 AUTO  
1     -1                 AUTO  
2     -1                 AUTO  
3      0                 AUTO  
4      1                 AUTO  

[5 rows x 27 columns]

In [72]:
df_xrt.dtypes

Model_Name                           object
Weights_Column                      float64
Offset_Column                       float64
Fold_Column                         float64
Fold_Assignment                      object
Stopping_Rounds                       int64
max_runtime_secs                      int64
Stopping_metric                      object
Stopping_tolerance                   object
Balance_Classes                        bool
Class_sampling_Factors              float64
Maxafter_Balance_Size                 int64
ntrees                                int64
max_depth                             int64
min_rows                              int64
nbins                                 int64
nbins_top_level                       int64
nbins_cats                            int64
seed                                 object
Sample_Rate                          object
Sample_rate_per_class               float64
Col_sample_rate_per_tree              int64
Col_sample_rate_Change_per_level

In [75]:
df_xrt= df_xrt.drop(columns=['Model_Name','Weights_Column','Offset_Column','Fold_Column','Fold_Assignment','Stopping_metric','histogram_type','categorical_encoding'])
df_xrt

Stopping_Rounds  max_runtime_secs Stopping_tolerance  Balance_Classes  \
0                0                 0        0.041415768            False   
1                0                 0        0.041415768            False   
2                0                 0        0.041415768            False   
3                0                 0        1.041415768            False   
4                0                 0        0.041415768            False   

   Class_sampling_Factors  Maxafter_Balance_Size  ntrees  max_depth  min_rows  \
0                     NaN                      5      16         20         1   
1                     NaN                      5      16         20         1   
2                     NaN                      5      13         20         1   
3                     NaN                      5      14         20         1   
4                     NaN                      5      17         20         1   

   nbins  nbins_top_level  nbins_cats          seed  Sample_Rate  \
0     20             1024        1024   4.84711E+18  0.632000029   
1     20             1024        1024   8.79434E+18  0.632000029   
2     20             1024        1024  -8.85259E+18  0.632000029   
3     20             1024        1024  -8.85259E+18  0.632000029   
4     20             1024        1024  -8.85259E+18  0.632000029   

   Sample_rate_per_class  Col_sample_rate_per_tree  \
0                    NaN                         1   
1                    NaN                         1   
2                    NaN                         1   
3                    NaN                         1   
4                    NaN                         1   

   Col_sample_rate_Change_per_level min_split_improvement  mtries  
0                                 1              1.00E-05      -1  
1                                 1              1.00E-05      -1  
2                                 1              1.00E+00      -1  
3                                 1              1.00E+00       0  
4                                 1              1.00E+00       1

In [77]:
ranges = []
for key,val in df_xrt.dtypes.items():
    print(key,val)
    if val == 'int64': 
         ranges.append([key,[df_xrt[key].min(),df_xrt[key].max()]])
    ##if val == 'float64':
      ##   ranges.append([key,[df_gbm[key].min(),df_gbm[key].max()]])
    if val == 'object':
         ranges.append([key,[df_xrt[key].min(),df_xrt[key].max()]])

Stopping_Rounds int64
max_runtime_secs int64
Stopping_tolerance object
Balance_Classes bool
Class_sampling_Factors float64
Maxafter_Balance_Size int64
ntrees int64
max_depth int64
min_rows int64
nbins int64
nbins_top_level int64
nbins_cats int64
seed object
Sample_Rate object
Sample_rate_per_class float64
Col_sample_rate_per_tree int64
Col_sample_rate_Change_per_level int64
min_split_improvement object
mtries int64


In [78]:
ranges

[['Stopping_Rounds', [0, 0]],
 ['max_runtime_secs', [0, 0]],
 ['Stopping_tolerance', ['0.041415768', '1.041415768']],
 ['Maxafter_Balance_Size', [5, 5]],
 ['ntrees', [13, 17]],
 ['max_depth', [20, 20]],
 ['min_rows', [1, 1]],
 ['nbins', [20, 20]],
 ['nbins_top_level', [1024, 1024]],
 ['nbins_cats', [1024, 1024]],
 ['seed', ['-8.85259E+18', '8.79434E+18']],
 ['Sample_Rate', ['0.632000029', '0.632000029']],
 ['Col_sample_rate_per_tree', [1, 1]],
 ['Col_sample_rate_Change_per_level', [1, 1]],
 ['min_split_improvement', ['1.00E+00', '1.00E-05']],
 ['mtries', [-1, 1]]]

|Hyperparameter| Minimum | Maximum |
|---|---|---|
|seed|-8.85259E+18 | 0.632000029|
|Stopping_Rounds|0| 0|
|max_runtime_secs|0| 0|
|Stopping_tolerance|0.041415768 |1.041415768|
|Maxafter_Balance_Size|5|5|
|ntrees|13| 17|
|max_depth|20| 20|
|min_rows|1|1|